File: exercism.io/go/scrabble-score/scrabble_score.go
package scrabble

import "strings"

// ScoreTable stores the score for each rune (in upper case)
var ScoreTable = map[rune]int{
	'A': 1,
	'E': 1,
	'I': 1,
	'O': 1,
	'U': 1,
	'L': 1,
	'N': 1,
	'R': 1,
	'S': 1,
	'T': 1,
	'D': 2,
	'G': 2,
	'B': 3,
	'C': 3,
	'M': 3,
	'P': 3,
	'F': 4,
	'H': 4,
	'V': 4,
	'W': 4,
	'Y': 4,
	'K': 5,
	'J': 8,
	'X': 8,
	'Q': 10,
	'Z': 10,
}

// Score calculates the scrabble score of the input string.
func Score(s string) int {
	sum := 0
	for _, c := range strings.ToUpper(s) {
		sum += ScoreTable[c]
	}
	return sum
}

File: exercism.io/dart/scrabble-score/lib/scrabble_score.dart
/// Letter-value map.
/// Letter                           Value
/// A, E, I, O, U, L, N, R, S, T       1
/// D, G                               2
/// B, C, M, P                         3
/// F, H, V, W, Y                      4
/// K                                  5
/// J, X                               8
/// Q, Z                               10
const Map<String, int> ScoreTable = const {
  'a': 1,
  'e': 1,
  'i': 1,
  'o': 1,
  'u': 1,
  'l': 1,
  'n': 1,
  'r': 1,
  's': 1,
  't': 1,
  'd': 2,
  'g': 2,
  'b': 3,
  'c': 3,
  'm': 3,
  'p': 3,
  'f': 4,
  'h': 4,
  'v': 4,
  'w': 4,
  'y': 4,
  'k': 5,
  'j': 8,
  'x': 8,
  'q': 10,
  'z': 10,
};

/// Calcuate the score of input string.
int score(String str) {
  int sum = 0;
  str = str.toLowerCase();
  for (int i = 0; i < str.length; i++) {
    sum += ScoreTable[str[i]];
  }
  return sum;
}
